packages/layout/src/aspect-ratio.ts

```diff
@@ -68,3 +68,13 @@ export const CAspectRatio = defineComponent({
     }
   },
 })
+
+/**
+ * Vue component used to cropping media (videos, images and maps)
+ * to a desired aspect ratio.
+ *
+ * @see Docs https://vue.chakra-ui.com/docs/layout/aspect-ratio
+ */
+export const CAspectRatio = CAspectRatioImpl as ComponentWithProps<
+  DeepPartial<AspectRatioProps>
+>
```

Comment:
Looking great here:

I think redeclaring the component seems a bit redundant. What do you think about defining the component and then coercing the type at the end or declaring the type before the variable?

**Example 1**
```ts
export const CAspectRatio: ComponentWithProps<AspectRatioProps> = defineComponent({
  name: 'CAspectRatio',
  // ...
})
```

**Example 2**
```ts
export const CAspectRatio = defineComponent({
  name: 'CAspectRatio',
  // ...
}) as ComponentWithProps<AspectRatioProps>
```

Reply:
> Looking great here:
> 
> I think redeclaring the component seems a bit redundant. What do you think about defining the component and then coercing the type at the end or declaring the type before the variable?
> 
> **Example 1**
> 
> ```ts
> export const CAspectRatio: ComponentWithProps<AspectRatioProps> = defineComponent({
>   name: 'CAspectRatio',
>   // ...
> })
> ```


oh i was wondering why the first syntax was not working:

![image](https://user-images.githubusercontent.com/342666/120213339-be824700-c23b-11eb-9523-b8a5f54f4542.png)

its bec i was not using `DeepPartial`. I was like type assertion works when i use with `as` keyword but not `const CBox: TYPE = defineComponent()` 
late night coding issues :D yea lets use this syntax. Thanks :D . i'll update the pr.